torch/_meta_registrations.py

```diff
@@ -143,3 +132,83 @@ def meta_repeat_interleave_Tensor(repeats, output_size=None):
             "cannot repeat_interleave a meta tensor without output_size"
         )
     return repeats.new_empty(output_size)
+
+@torch.library.impl(meta_lib, "index.Tensor")
+def meta_index_Tensor(self, indices):
+    # aten::index is the internal advanced indexing implementation
+    check(len(indices) <= self.ndim, f"too many indices for tensor of dimension {self.ndim} (got {len(indices)})")
```

Comment:
The second argument should be a lambda?
I think we shñould make it accept a `str` as well, as I've seen it used wrong in other places, like `meta_adaptive_avg_pool3d`.

Reply:
Making it a lambda is important because without it the string formatting opcodes happen always whereas as a lambda they only happen when you error. An easy fix barring type checking probably is to just assert that the second argument is callable.